File: quiz_app_flutter/lib/domain/mapper/quiz_set_mapper.dart
import '../../data/model/dashboard/dashboard_dto.dart';
import '../entities/quiz_set/quiz_set_data.dart';

extension DashboardItemDtoMapper on ItemDto {
  QuizSetData toQuizSetDomain() {
    return QuizSetData(
      total: total,
      title: title,
      topic: topic,
      description: description,
      sectionId: sectionId,
      sections: sections.map((e) => e.toQuizSetDomain()).toList(),
    );
  }
}

extension DashboardSectionDtoMapper on SectionDto {
  QuizSetItem toQuizSetDomain() {
    return QuizSetItem(title: title, description: description, position: position, fileName: fileName);
  }
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/data/mapper/QuizSetMapper.kt
package com.tnm.quizappkmp.data.mapper

import com.tnm.quizappkmp.data.dto.DashboardDto
import com.tnm.quizappkmp.domain.model.quizSets.QuizSetData

fun DashboardDto.ItemDto.toQuizSetDomain() = QuizSetData(
    total = total,
    title = title,
    topic = topic,
    description = description,
    sectionId = sectionId,
    sections = sections.map { it.toQuizSetDomain() }
)

fun DashboardDto.SectionDto.toQuizSetDomain() = QuizSetData.SectionItem(
    title = title,
    description = description,
    position = position,
    fileName = fileName,
)